src/libraries/Core/Microsoft.Agents.Core/HeaderPropagation/HeaderPropagationContext.cs

```diff
@@ -89,7 +89,17 @@ private static Dictionary<string, StringValues> FilterHeaders(Dictionary<string,
 #endif
                     break;
                 case HeaderPropagationEntryAction.Append when headerExists:
-                    StringValues newValue = requestHeader.Concat(header.Value).ToArray();
+                    StringValues newValue;
+                    if (string.Equals(header.Key, "user-agent", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // Special handling for User-Agent to concatenate values with a space.
+                        newValue = new StringValues(string.Join(" ", [.. requestHeader, header.Value]));
```

Comment:
nit: we could avoid creating another variable and assign it to result[header.Key]